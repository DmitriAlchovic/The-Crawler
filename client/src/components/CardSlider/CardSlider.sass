.cardSliderContainer
    height: 271px
    margin: 97px 36px 107px 36px

.sliderHeader
    display: flex
    justify-content: space-between
    margin: 0 120px

.sliderBody
    display: flex
    justify-content: space-between
    align-items: center

    .arrow-left
        cursor: pointer

    .arrow-right
        rotate: 180deg
        margin-left: 21px
        cursor: pointer

.card-container
    width: 126rem
    display: flex
    position: relative
    justify-content: center
    min-width: auto
    min-height: auto

.category-selector
    display: flex

    .active-category
        cursor: pointer
        color: #3bb77e
        font-size: 16px
        margin-right: 19px

    .category
        cursor: pointer
        font-size: 16px
        margin-right: 19px

.arrow-container
    width: 46px
    height: 46px