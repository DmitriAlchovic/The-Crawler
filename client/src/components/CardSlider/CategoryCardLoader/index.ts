import CategoryCardLoader from './CategoryCardLoader';

export default CategoryCardLoader;
