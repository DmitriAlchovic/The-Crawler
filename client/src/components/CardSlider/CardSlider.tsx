/* eslint-disable operator-linebreak */
import React from 'react';
import './CardSlider.sass';
import arrow from '../../assets/arrow.svg';
import { Category } from '../../store/reducers/categorySlice';

interface CardSliderProps {
  children: React.ReactNode;
  index: number;
  title: string;
  selection: Category[] | null;
  activeCategory: string;
  sliderHandler(step: number): void;
  categoryHandler(category: string): void;
  sliderSize: number;
  cardsCount: number;
}

export default function CardSlider({
  children,
  index,
  title,
  selection,
  activeCategory,
  sliderHandler,
  categoryHandler,
  sliderSize,
  cardsCount,
}: CardSliderProps) {
  return (
    <div className="cardSliderContainer">
      <div className="sliderHeader">
        <div>
          <h1>{title}</h1>
        </div>
        <div className="category-selector">
          <div
            className={
              activeCategory === 'All' ? 'active-category' : 'category'
            }
            role="button"
            tabIndex={-1}
            onKeyDown={() => {}}
            onClick={() => {
              categoryHandler('All');
            }}
          >
            All
          </div>
          {selection &&
            selection.map(({ name }) => (
              <div
                className={
                  activeCategory === name ? 'active-category' : 'category'
                }
                key={name}
                role="button"
                tabIndex={-1}
                onKeyDown={() => {}}
                onClick={() => {
                  categoryHandler(name);
                }}
              >
                {name}
              </div>
            ))}
        </div>
      </div>
      <div className="sliderBody">
        <div className="arrow-container">
          {index - sliderSize > 0 && (
            <div
              className="arrow-left"
              role="button"
              tabIndex={-1}
              onKeyDown={() => {}}
              onClick={() => {
                sliderHandler(-1);
              }}
            >
              <img alt="left arrow" src={arrow} />
            </div>
          )}
        </div>
        <div className="card-container">{children}</div>
        <div className="arrow-container">
          {cardsCount >= index && (
            <div
              className="arrow-right"
              role="button"
              tabIndex={-1}
              onKeyDown={() => {}}
              onClick={() => {
                sliderHandler(1);
              }}
            >
              <img alt="right arrow" src={arrow} />
            </div>
          )}
        </div>
      </div>
    </div>
  );
}
