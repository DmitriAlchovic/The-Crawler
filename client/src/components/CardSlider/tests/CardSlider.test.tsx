import React from 'react';
import { render } from '@testing-library/react';
import * as reduxHooks from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CardSlider from '../CardSlider';

jest.mock('react-redux');

describe('CardSlider', () => {
  it('should create empty component', () => {
    jest.spyOn(reduxHooks, 'useSelector').mockReturnValue('');
    const component = render(
      <BrowserRouter>
        <CardSlider
          index={0}
          title=""
          selection={null}
          activeCategory=""
          sliderHandler={() => {}}
          categoryHandler={() => {}}
          sliderSize={0}
          cardsCount={0}
        >
          <div />
        </CardSlider>
      </BrowserRouter>,
    );

    expect(component).toMatchSnapshot();
  });
});
