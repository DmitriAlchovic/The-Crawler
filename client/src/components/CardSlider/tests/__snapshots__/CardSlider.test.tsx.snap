// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSlider should create empty component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardSliderContainer"
      >
        <div
          class="sliderHeader"
        >
          <div>
            <h1 />
          </div>
          <div
            class="category-selector"
          >
            <div
              class="category"
              role="button"
              tabindex="-1"
            >
              All
            </div>
          </div>
        </div>
        <div
          class="sliderBody"
        >
          <div
            class="arrow-container"
          />
          <div
            class="card-container"
          >
            <div />
          </div>
          <div
            class="arrow-container"
          >
            <div
              class="arrow-right"
              role="button"
              tabindex="-1"
            >
              <img
                alt="right arrow"
                src="arrow.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardSliderContainer"
    >
      <div
        class="sliderHeader"
      >
        <div>
          <h1 />
        </div>
        <div
          class="category-selector"
        >
          <div
            class="category"
            role="button"
            tabindex="-1"
          >
            All
          </div>
        </div>
      </div>
      <div
        class="sliderBody"
      >
        <div
          class="arrow-container"
        />
        <div
          class="card-container"
        >
          <div />
        </div>
        <div
          class="arrow-container"
        >
          <div
            class="arrow-right"
            role="button"
            tabindex="-1"
          >
            <img
              alt="right arrow"
              src="arrow.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Dropdown should create empty component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cardSliderContainer"
      >
        <div
          class="sliderHeader"
        >
          <div>
            <h1 />
          </div>
          <div
            class="category-selector"
          >
            <div
              class="category"
              role="button"
              tabindex="-1"
            >
              All
            </div>
          </div>
        </div>
        <div
          class="sliderBody"
        >
          <div
            class="arrow-container"
          />
          <div
            class="card-container"
          >
            <div />
          </div>
          <div
            class="arrow-container"
          >
            <div
              class="arrow-right"
              role="button"
              tabindex="-1"
            >
              <img
                alt="right arrow"
                src="arrow.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cardSliderContainer"
    >
      <div
        class="sliderHeader"
      >
        <div>
          <h1 />
        </div>
        <div
          class="category-selector"
        >
          <div
            class="category"
            role="button"
            tabindex="-1"
          >
            All
          </div>
        </div>
      </div>
      <div
        class="sliderBody"
      >
        <div
          class="arrow-container"
        />
        <div
          class="card-container"
        >
          <div />
        </div>
        <div
          class="arrow-container"
        >
          <div
            class="arrow-right"
            role="button"
            tabindex="-1"
          >
            <img
              alt="right arrow"
              src="arrow.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
